### PR TITLE
Get latest functional gauge addr from subgraph

### PR DESCRIPTION
**Pull request by petrovska-petro**
Tackles #1034

### CODE REVIEW

File: great_ape_safe/ape_api/balancer.py [hunk 1]
```diff
@@ -274,7 +290,7 @@ def _deposit_and_stake(self, pool, request, stake=True, destination=None):
     def stake(self, pool, mantissa, destination=None, dusty=False):
         pool_id = pool.getPoolId()
         destination = self.safe if not destination else destination
-        gauge_address = self.gauge_factory.getPoolGauge(pool)
+        gauge_address = self.get_preferential_gauge(pool.lower())
```
Comment (shuklaayush) [inline great_ape_safe/ape_api/balancer.py:293]: I guess we should change all other references to `gauge_factory.getPoolGauge` as well

File: great_ape_safe/ape_api/balancer.py [hunk 2]
```diff
@@ -108,6 +114,16 @@ def get_pool_data(self, update_cache=False):
             data = json.load(f)
             return data
 
+    def get_preferential_gauge(self, pool_id):
+        r = requests.post(
+            self.gauges_subgraph,
+            json={"query": pool_preferential_gauge, "variables": {"pool_id": pool_id}},
```
Comment (shuklaayush) [inline great_ape_safe/ape_api/balancer.py:120]:
```suggestion
    def get_preferential_gauge(self, pool):
        r = requests.post(
            self.gauges_subgraph,
            json={"query": pool_preferential_gauge, "variables": {"pool_address": str(pool).lower()}},
```

File: great_ape_safe/ape_api/balancer.py (hunk 1 above)
Comment (shuklaayush) [inline great_ape_safe/ape_api/balancer.py:293]:
```suggestion
        gauge_address = self.get_preferential_gauge(pool)
```

File: great_ape_safe/ape_api/helpers/balancer/queries.py [hunk 3]
```diff
@@ -11,3 +11,12 @@
           }
         }
         """
+pool_preferential_gauge = """
+        query($pool_id: ID!) {
+          pool(id: $pool_id) {
```
Comment (shuklaayush) [inline great_ape_safe/ape_api/helpers/balancer/queries.py:16]:
```suggestion
        query($pool_address: ID!) {
          pool(id: $pool_address) {
```

File: great_ape_safe/ape_api/balancer.py (hunk 1 above)
Comment (petrovska-petro) [reply great_ape_safe/ape_api/balancer.py]: resolved in f2f4c7cb07bf0477806711c775db4982393a4a5e

**Review by shuklaayush — APPROVED** [review]

**Review by gosuto-inzasheru — APPROVED** [review]
looks good

only thing is that for unstaking we now will have to manually get gauge from legacy factory and call withdraw directly

**Pull request merged**